<launch>
    <arg name="sim" default="true" doc="sim or real" />

    <!-- accumulate range messages to pointcloud2 messages -->
    <node name="range_to_pointcloud2_converter" pkg="iiwa_workcell_sensor" type="range_to_pointcloud2_converter.py" output="screen">
        <rosparam param="topics_in">[sensor_data]</rosparam>
        <param name="topic_out" value="sensor_points"/>
        <param name="frame_out" value="world"/>
    </node>

    <group if="$(arg sim)">
        <!-- Publish fake range data to simulate sensors -->
        <node name="range_faker" pkg="iiwa_workcell_sensor" type="range_faker.py" output="screen">
            <rosparam param="frames_in" >[sensor_2_front,sensor_2_left,sensor_2_back,sensor_2_right,sensor_4_front,sensor_4_left,sensor_4_back,sensor_4_right,sensor_6_front_left,sensor_6_front_right]</rosparam>
            <param name="topic_out" value="sensor_data"/>
        </node>
    </group>
    <group unless="$(arg sim)">
        <!-- Publish real range messages (note that this must run on the ros master) -->
        <include file="$(find rosserial_server)/launch/socket.launch" />
    </group>
</launch>
